app/controllers/assignments_controller.rb

```diff
@@ -8,35 +14,45 @@ def create
 
       if shift_assignment_service.perform
         flash[:success] = shift_assignment_service.success
+
       else
         flash[:danger] = shift_assignment_service.errors
       end
     end
 
-    redirect_to home_path
+    redirect_to assignments_new_path
   end
 
   # rubocop:disable Metrics/AbcSize
   # :reek:TooManyStatements
   # :reek:NilCheck
   # :reek:DuplicateMethodCall
   def show
-    if !params.has_key?(:range1) && !params.has_key?(:range2)
-      flash[:danger] = 'Missing date input!'
-      redirect_to home_path
-    end
-
-    @range = date_range(params[:range1], params[:range2])
+    @range = if !params.has_key?(:range1) && !params.has_key?(:range2)
+               Time.zone.today.at_beginning_of_day..Time.zone.today.at_end_of_day
+             else
+               date_range(params[:range1], params[:range2])
+             end
 
     if @range.max == nil
       flash[:danger] = 'Invalid date range!'
-      redirect_to home_path
+      redirect_to assignments_show_path
     end
 
     @shifts = retrieve_shift_in_range(@range)
   end
   # rubocop:enable Metrics/AbcSize
 
+  # Load Javascript to re-render partial
+  def search
```

Comment:
actually, for filtering like this, we usually prefer to reuse the index action rather than using ajax, so that the parameters are included in the GET url, which makes it easier to bookmark the frequently used filters, or to share the link with others

Reply:
Fixed